From 72f796280737414f38d8a48a47be8b655a31c80d Mon Sep 17 00:00:00 2001
From: Josselin Mouette <joss@debian.org>
Date: Sat, 10 Apr 2010 12:36:31 +0200
Subject: [PATCH] Remove autoscripts for GConf

Starting from 2.28.1-2, the gconf2 package uses triggers which make the
maintainer scripts obsolete.
---
 autoscripts/postinst-gconf          |    3 ---
 autoscripts/postinst-gconf-defaults |    3 ---
 autoscripts/postrm-gconf-defaults   |    3 ---
 autoscripts/prerm-gconf             |    3 ---
 dh_gconf                            |   11 +++--------
 5 files changed, 3 insertions(+), 20 deletions(-)
 delete mode 100644 autoscripts/postinst-gconf
 delete mode 100644 autoscripts/postinst-gconf-defaults
 delete mode 100644 autoscripts/postrm-gconf-defaults
 delete mode 100644 autoscripts/prerm-gconf

diff --git a/autoscripts/postinst-gconf b/autoscripts/postinst-gconf
deleted file mode 100644
index eec1d51..0000000
--- a/autoscripts/postinst-gconf
+++ /dev/null
@@ -1,3 +0,0 @@
-if [ "$1" = "configure" ]; then
-	gconf-schemas --register #SCHEMAS#
-fi
diff --git a/autoscripts/postinst-gconf-defaults b/autoscripts/postinst-gconf-defaults
deleted file mode 100644
index eff244a..0000000
--- a/autoscripts/postinst-gconf-defaults
+++ /dev/null
@@ -1,3 +0,0 @@
-if [ "$1" = "configure" ] && which update-gconf-defaults >/dev/null 2>&1; then
-	update-gconf-defaults #OPT#
-fi
diff --git a/autoscripts/postrm-gconf-defaults b/autoscripts/postrm-gconf-defaults
deleted file mode 100644
index b15de67..0000000
--- a/autoscripts/postrm-gconf-defaults
+++ /dev/null
@@ -1,3 +0,0 @@
-if which update-gconf-defaults >/dev/null 2>&1; then
-	update-gconf-defaults #OPT#
-fi
diff --git a/autoscripts/prerm-gconf b/autoscripts/prerm-gconf
deleted file mode 100644
index 6a18816..0000000
--- a/autoscripts/prerm-gconf
+++ /dev/null
@@ -1,3 +0,0 @@
-if [ "$1" = remove ] || [ "$1" = upgrade ]; then
-	gconf-schemas --unregister #SCHEMAS#
-fi
diff --git a/dh_gconf b/dh_gconf
index 514aae7..bf9f398 100755
--- a/dh_gconf
+++ b/dh_gconf
@@ -72,17 +72,14 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 	if ($mandatory ne '') {
 		doit("mkdir","-p","$tmp/usr/share/gconf/mandatory");
 		doit("install","-p","-m644",$mandatory,"$tmp/usr/share/gconf/mandatory/${priority}_$package");
-		autoscript($package,"postinst","postinst-gconf-defaults","s%#OPT#%--mandatory%");
-		addsubstvar($package, "misc:Depends", "gconf2 (>= 2.24.0-5)");
+		addsubstvar($package, "misc:Depends", "gconf2 (>= 2.28.1-2)");
 		$gconf_dep = 1;
 	}
 	my $defaults = pkgfile($package,"gconf-defaults");
 	if ($defaults ne '') {
 		doit("mkdir","-p","$tmp/usr/share/gconf/defaults");
 		doit("install","-p","-m644",$defaults,"$tmp/usr/share/gconf/defaults/${priority}_$package");
-		autoscript($package,"postinst","postinst-gconf-defaults","s%#OPT#%%");
-		autoscript($package,"postrm","postrm-gconf-defaults","s%#OPT#%%");
-		addsubstvar($package, "misc:Depends", "gconf2 (>= 2.12.1-1)") unless $gconf_dep;
+		addsubstvar($package, "misc:Depends", "gconf2 (>= 2.28.1-2)") unless $gconf_dep;
 		$gconf_dep = 1;
 	}
 
@@ -100,9 +97,7 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 		# Get a list of the schemas
 		my $schemas = `find $new_schemas_dir -type f -name \\*.schemas -printf '%P '`;
 		if ($schemas ne '') {
-			autoscript($package,"postinst","postinst-gconf","s%#SCHEMAS#%$schemas%");
-			autoscript($package,"prerm","prerm-gconf","s%#SCHEMAS#%$schemas%");
-			addsubstvar($package, "misc:Depends", "gconf2 (>= 2.10.1-2)") unless $gconf_dep;
+			addsubstvar($package, "misc:Depends", "gconf2 (>= 2.28.1-2)") unless $gconf_dep;
 		}
 	}
 }
-- 
1.7.0.4

